src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,53 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: String) -> Result<(), String> {
+    let mut devfile = OpenOptions::new()
+        .read(true)
+        .open(&devpath)
+        .map_err(|the_io_error| format!("Error opening device: {}", the_io_error))?;
+
+    let bda = BDA::load(&mut devfile)
+        .map_err(|bda_load_error| format!("BDA detected but error found: {}", bda_load_error))?
+        .ok_or_else(|| "No Stratis BDA metadata found".to_string())?;
+    println!("{:#?}", bda);
+
+    devfile.seek(SeekFrom::Start(0)).unwrap();
+
+    let loaded_state = bda
+        .load_state(&mut devfile)
+        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?;
+    println!("State JSON data:");
```

Comment:
Please add a few more newlines, to group the actions better, like just after this ```println!```

Reply:
I'm not entirely sure what you mean (though I also might have missed this since I was editing in an 80-column terminal).

Would it be better to have a newline right before every `let <variable> = <function_call()>.map[_err]()` set of lines?

Here's an example of what it would look like:
```
fn run(devpath: &str) -> Result<(), String> {
    let mut devfile = OpenOptions::new()
        .read(true)
        .open(&devpath)
        .map_err(|the_io_error| format!("Error opening device: {}", the_io_error))?;

    let bda = BDA::load(&mut devfile)
        .map_err(|bda_load_error| format!("BDA detected but error found: {}", bda_load_error))?
        .ok_or_else(|| "No Stratis BDA metadata found".to_string())?;
    println!("{:#?}", bda);

    devfile.seek(SeekFrom::Start(0)).unwrap();

    let loaded_state = bda
        .load_state(&mut devfile)
        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?;

    println!("State JSON data:");
    let state_json: Value = serde_json::from_slice(&loaded_state.unwrap())
        .map_err(|extract_err| format!("Error during state JSON extract: {}", extract_err))?;

    let state_json_pretty: String = serde_json::to_string_pretty(&state_json)
        .map_err(|parse_err| format!("Error during state JSON parse: {}", parse_err))?;

    println!("{}", state_json_pretty);
    Ok(())
}
```

Reply:
Well, the way I'm thinking of it, is that the code takes a few steps to get some data, and then it displays it. Then there's a rinse, repeat sort of behavior. So after any ```println!()```, I'ld put in a newline, indicating that that action was completed and that a new action is commencing, which will result in more metadata being displayed, or an error return, of course.

Reply:
I pushed a commit to add newlines (hopefully in the right places); let me know how it looks.

Reply:
I think it looks good, I'm resolving it.